src/views/components/artworkDetails.js

```diff
@@ -21,50 +21,33 @@ export default function ArtworkDetails({navigation}) {
     price: navigation.getParam('price'),
     quantity: 1,
   };
-  // let arr; 
-  // const getData = async () => {
-  //   try {
-  //     await AsyncStorage.getItem('save_data')
-  //       .then(jsonValue => {
-  //         return jsonValue != null ? JSON.parse(jsonValue) : null;
-  //       })
-  //       .then(res => {
-  //         arr = res;
-  //         if (arr != null) { 
-  //           arr.push(items);
-  //         } else {
-  //           arr = [];
-  //           arr.push(items); 
-  //         }
-  //       });
-  //   } catch (e) {
-  //     console.log(e);
-  //   }
-  // };
 
-  let arr;
+   let arr;
   const getData = async () => {
     try {
-      const value = await AsyncStorage.getItem('save_data');
-      const jsonValue= value != null ? JSON.parse(value) : null;
-      console.log(jsonValue)
-      arr = jsonValue;
-      if (arr !== null) { 
-        arr.push(items);
-      } else {
-        arr = [];
-        arr.push(items);
-      }
-      console.log("data",arr);
+      await AsyncStorage.getItem('save_data')
```

Comment:
Move Async Storage-related operations to AsyncStorageService file. export it as a function that returns parsed JSON value.